library/src/main/java/com/novoda/downloadmanager/lib/DownloadTask.java

```diff
@@ -449,6 +453,12 @@ private void checkPausedOrCanceled() throws StopRequestException {
         }
     }
 
+    private void checkClientRules() throws StopRequestException {
+        if (!downloadReadyChecker.clientAllowsToDownload(originalDownloadBatch)) {
```

Comment:
Oh I see, with your little more explanation now is clearer. 

That is correct, we send the original one. 

Now that you mention it, if a client checks for free space and we don't do that while downloading. If another application is downloading something, we would potentially continue until consuming all free space. Is that correct? 

I'm not sure of querying the full object as for performance impact. 

Considering my precios question, a client could check free space before downloading. If that was the case, not querying the full batch every time wouldn't change current behaviour. So we could go as we are now and as agreed move directly to the new column implementation.